var/spack/repos/builtin/packages/dd4hep/package.py

```diff
@@ -102,6 +122,27 @@ def cmake_args(self):
             "-DBoost_NO_BOOST_CMAKE=ON",
             "-DPYTHON_EXECUTABLE={0}".format(spec['python'].command.path),
         ]
+        subpackages = []
+        if spec.satisfies('+ddg4'):
+            subpackages += ['DDG4']
+        if spec.satisfies('+ddcond'):
+            subpackages += ['DDCond']
+        if spec.satisfies('+ddcad'):
+            subpackages += ['DDCAD']
+        if spec.satisfies('+ddrec'):
+            subpackages += ['DDRec']
+        if spec.satisfies('+dddetectors'):
+            subpackages += ['DDDetectors']
+        if spec.satisfies('+ddalign'):
+            subpackages += ['DDAlign']
+        if spec.satisfies('+dddigi'):
+            subpackages += ['DDDigi']
+        if spec.satisfies('+ddeve'):
+            subpackages += ['DDEve']
+        if spec.satisfies('+utilityapps'):
+            subpackages += ['UtilityApps']
```

Comment:
I think this code could become less redundant by using a variant -> module dictionary and iterating over it.

Reply:
I was thinking about it,  it's probably on the threshold where that would actually save some LOC. Since the number of DD4hep subpackages won't scale up indefinitely (I'm hoping, at least ...) I'd tend to just leave it like this.